ArcMovies/ArcMovies/ViewModels/MainPageViewModel.cs

```diff
@@ -0,0 +1,134 @@
+﻿using Plugin.Connectivity;
+using Prism.Commands;
+using Prism.Navigation;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using ArcMovies.Helpers;
+using ArcMovies.Models;
+using Xamarin.Forms;
+
+namespace ArcMovies.ViewModels
+{
+    public class MainPageViewModel : BaseViewModel
+    {        
+        private int _currentPage = 1;
+        private int _totalPage = 0;
+
+        private bool _isConnected;
+        public bool IsConnected
+        {
+            get => _isConnected;
+            set => SetProperty(ref _isConnected, value);
+        }
+
+        private List<Genre> _genres;        
+        public ObservableRangeCollection<Movie> Movies { get; set; }
+
+        public DelegateCommand LoadUpcomingMoviesCommand { get; }
+        public DelegateCommand ShowSearchMoviesCommand { get; }        
+        public DelegateCommand<Movie> ShowMovieDetailCommand { get; }
+        public DelegateCommand<Movie> ItemAppearingCommand { get; }
+
+        private readonly INavigationService _navigationService;
+        public MainPageViewModel(INavigationService navigationService)
+        {
+            Title = "Upcoming Movies";
+            this._navigationService = navigationService;
+            Movies = new ObservableRangeCollection<Movie>();
+
+            LoadUpcomingMoviesCommand = new DelegateCommand(async () => await ExecuteLoadUpcomingMoviesCommand().ConfigureAwait(false));
+            ShowSearchMoviesCommand = new DelegateCommand(async () => await ExecuteShowSearchMoviesCommand().ConfigureAwait(false));
+            ShowMovieDetailCommand = new DelegateCommand<Movie>(async (Movie movie) => await ExecuteShowMovieDetailCommand(movie).ConfigureAwait(false));
+            ItemAppearingCommand = new DelegateCommand<Movie>(async (Movie movie) => await ExecuteItemAppearingCommand(movie).ConfigureAwait(false));
+
+            LoadUpcomingMoviesCommand.Execute();
+        }        
+
+        private async Task ExecuteLoadUpcomingMoviesCommand()
+        {
+            IsConnected = CrossConnectivity.Current.IsConnected;
+
+            if (IsBusy || !IsConnected)
+                return;
+
+            IsBusy = true;
+
+            try
+            {
+                Movies.Clear();
+                _currentPage = 1;
+                await LoadMoviesAsync(_currentPage, Enums.MovieCategory.Upcoming).ConfigureAwait(false);
+            }
+            finally
+            {
+                IsBusy = false;
+            }
+        }
+
+        private async Task ExecuteShowSearchMoviesCommand()
+        {            
+            await _navigationService.NavigateAsync("SearchMoviesPage").ConfigureAwait(false);
```

Comment:
`"SearchMoviesPage"` is a weak reference to a page name. I recommend making use of `nameof(SearchMoviesPage)` so you're safe in case the page name changes.